Client/Frontend/Browser/Tab.swift

```diff
@@ -134,7 +134,8 @@ class Tab: NSObject {
     }
 
     private func updateAppState() {
-        self.appStateDelegate?.appDidUpdateState(AppState(ui:.Tab(tabState: self.tabState)))
+        let state = mainStore.updateState(.Tab(tabState: self.tabState))
+        self.appStateDelegate?.appDidUpdateState(state)
```

Comment:
it feels like each delegate should register with the main store for updates, not with each component, so the store notifies a state change. I know that is a bit more Fluxy than Reduxy but might fits the current 'in between architectures' model a little better.